interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Controller/CarecoordinationController.php

```diff
@@ -909,4 +914,152 @@ private function getRevAndApproveAuditArray($audit_master_id, $table_name)
+                    $shouldDeleteIndex = true;
+                } else {
+                    $patientCountHash[$patientNameIndex] += 1;
+                }
+            } else {
+                $shouldDeleteIndex = true;
+            }
+
+            // TODO: fire off an event with the patient name, current index, file name, and zip archive object
+            // we can filter on whether we should keep this and let module writers do their own thing if they want to
+            // retain any of the documents or not for their own custom processing.
+            if ($shouldDeleteIndex) {
+                $z->deleteIndex($i);
+            }
+        }
+        $z->close();
+    }
+
+    private function printZipContents($zipLocation)
+    {
+        $z = new \ZipArchive();
+        $z->open($zipLocation);
+        for ($i = 0; $i < $z->numFiles; $i++) {
+            $stat = $z->statIndex($i);
+            (new SystemLogger())->error("File in zip is " . $stat['name']);
+        }
+        $z->close();
+    }
+
+    private function importZipUpload($request)
+    {
+
+
+        // our file structure is
+        // import_name / patient_name / ccda.xml
+        // import_name / patient_name / ccda.html
+        // import_name / patient_name / ccda.xsl
+
+        // we will need to have these limit options configurable but we have to be careful to
+        // we will limit our docsToImport to 500
+        // we will limit our patientsToImport to 500
+
+        $z = new \ZipArchive();
+        $tmpFile = reset($_FILES);
+        $tmpFileName = $tmpFile['tmp_name'];
+        $this->printZipContents($tmpFileName);
+
+        // make sure we only have our documents folder and our ccda file
+        $this->sanitizeZip($tmpFileName);
+        $z->open($tmpFileName);
+        $category_details          = $this->getCarecoordinationTable()->fetch_cat_id('CCDA');
+        $catId = $category_details[0]['id'] ?? null;
+        if (empty($catId)) {
+            throw new \RuntimeException("Could not find document category id for category of CCDA");
+        }
+        $auditMasterRecordByPatients = [];
+        for ($i = 0; $i < $z->numFiles; $i++) {
+            $stat = $z->statIndex($i);
+            // explode and make sure we have our three parts
+            // our max directory structure is 4... anything more than that and we will bail
+            $fileComponents = explode("/", $stat['name'], 5);
+            $componentCount = count($fileComponents);
+
+            // now we need to do our document import for our ccda for this patient
+            if ($componentCount == 3) {
+                // let's process the ccda
+                $file_name = basename($stat['name']);
+
+                $pid = '00';
+                $ob = new Document();
+                $contents = $z->getFromIndex($i);
+                $ret = $ob->createDocument($pid, $catId, $file_name, 'text/xml', $contents, '', 1, 0);
```

Comment:
would drop the last 3 parameters of this function call since are just the default settings